Assets/MRTK/Core/AssemblyInfo.cs

```diff
@@ -8,3 +8,5 @@
 
 [assembly: AssemblyProduct("Microsoft® Mixed Reality Toolkit")]
 [assembly: AssemblyCopyright("Copyright © Microsoft Corporation")]
+
+[assembly: UnityEngine.Scripting.Preserve]
```

Comment:
Not sure how I feel about marking so many of our assemblies with this tag. Our existing link.xml might be a contributor to reported increased build times. https://github.com/microsoft/MixedRealityToolkit-Unity/issues/5273#issuecomment-513974004

Reply:
Can we just mark types that are used via `SystemType` serialization? It feels like that's the problem case.

Reply:
I can do that. It will likely be the services, providers and extensions (which is the majority of those changed here).

Reply:
That said, a pruning of the list _is_ warranted here

Reply:
@keveleigh, do you think it is worth marking the _classes_ vs marking the entire assemblies?

Reply:
Added a custom preserve attribute that supports inheritance. this allows us to decorate only the base class(es) to perform strategic, targeted preservation